### DISCUSSION TITLE
Best way to update bindings that use x:reference as command parameters

### DISCUSSION THREAD

**Discussion by Axemasta** [dotnet/maui#26028] [Q&A]
I've been upgrading my app to .NET 9 and I'm going through updating any of the xaml compile warnings that have appeared in my project.

I have the following scenario that I'm unsure on the best way to upgrade after reading the docs I'm not sure I understand how I need to change my xaml...

### Xaml:
```xml
<controls:SelectableEntryHierarchyView
    Entries="{Binding Entries}"
    x:Name="EntryHierarchyView" />

<controls:FooterView>
    <Button
        Command="{Binding LogEntryCommand}"
        CommandParameter="{Binding Source={x:Reference EntryHierarchyView}, Path=SelectedEntry}"
        Text="Log Entry" />
</controls:FooterView>
```

### Command
```csharp
[RelayCommand]
private async Task LogEntry(EntryDescriptionHierarchy? entryDescriptionHierarchy)
{
    if (entryDescriptionHierarchy is null)
    {
        return;
    }

    // Do something
}
```

If I add the `x:DataType` to my command parameter:
```xaml
CommandParameter="{Binding Source={x:Reference EntryHierarchyView}, Path=SelectedEntry, x:DataType=controls:SelectableEntryHierarchyView}"
```
I get the following warning
<img width="834" alt="image" src="https://github.com/user-attachments/assets/7f7a8237-4bb2-4e3b-a0ba-0408e41666f5">

I'm not sure if this way of using xaml is recommended anymore. Could anyone explain whether this is the correct approach and offer suggestions?

The reason I'm using `x:Reference` here is I can pass the object directly to my command as opposed to having my list bind to a property in my viewmodel, then my command have to check it there. In my opinion this is a cleaner approach of setting up my UI, is this approach still valid?

**Comment by Axemasta** [comment]
@simonrozsival appreciate you're probably quite busy, but would you be able to clarify my question given the scenario above fro the new xaml compilation engine?

**Reply by Axemasta** [reply]
The same error is present for `CollectionView`:
```xml
<CollectionView
    ItemTemplate="{StaticResource DownloadedLogbookCard}"
    ItemsSource="{Binding DownloadedLogbooks}"
    SelectedItem="{Binding SelectedLogbook}"
    SelectionChangedCommand="{Binding LogbookSelectedCommand}"
    SelectionChangedCommandParameter="{Binding Path=SelectedItem, Source={RelativeSource Self}, x:DataType=CollectionView}"
    SelectionMode="Single">
```

**Reply by simonrozsival** [reply]
Hi @Axemasta. That warning doesn't seem right, this might be a bug. Your code should work. Could you share a repro app that I could use for further investigation?

**Reply by Axemasta** [reply]
I've made a repro of both scenarios (button command param & collection view) [here](https://github.com/Axemasta/MauiRepros/tree/main/src/ReferenceCommandParameterRepro).

On both source bindings with `x:DataType` i see the warning reported above:
<img width="1507" alt="image" src="https://github.com/user-attachments/assets/c6cf51d1-4f19-4968-a9f7-4c2ef3c92a3d">

<img width="1373" alt="image" src="https://github.com/user-attachments/assets/6282c7ca-694b-4422-935f-0b49cc7312a8">

It did occur to me that this could be a rider issue because its not a warning reported by the cli, however it doesn't hurt to check whether this is the correct way to be doing things!

**Reply by maonaoda** [reply]
Which IDE are you using? VSCode？

---

Will there be a warning if you write it like this?
```
            <Button
                Command="{Binding ReportCommand}"
                CommandParameter="{Binding Source={x:Reference DatePicker}, Path=Date, x:DataType={DatePicker}}"
                Text="Report Date" />
```

**Reply by simonrozsival** [reply]
I'm not getting the same warning in build logs or in VS Code. Are you getting that warning just in the IDE? Can you build and run that app?

**Reply by maonaoda** [reply]
It turns out to be Rider

![image](https://github.com/user-attachments/assets/82620114-aa8f-405a-903d-79bae67b0735)

**Reply by Axemasta** [reply]
> I'm not getting the same warning in build logs or in VS Code. Are you getting that warning just in the IDE? Can you build and run that app?

Yes the app builds, runs and behaves as expected. It's probably just a rider issue by the looks of it!

**Comment by stephenquan** [comment]
I understand the context in which you're using the term "rider issue", but I hate to ask the obvious: what is its definition and, possibly, the etymology?

**Reply by ahoke-cr** [reply]
Rider is the name of an IDE by JetBrains, an alternative some prefer to Visual Studio.
